mlflow/server/js/src/Actions.js

```diff
@@ -185,16 +185,36 @@ export const getUUID = () => {
  * Jquery's ajax promise is a bit weird so I chose to create a new Promise which resolves and
  * rejects using the ajax callbacks `success` and `error`.
  */
-const wrapDeferred = (deferred, data) => {
+export const wrapDeferred = (deferred, data, timeLeftMs = 60000, sleepMs = 1000) => {
   return new Promise((resolve, reject) => {
     deferred({
       data,
-      success: response => resolve(response),
-      error: xhr => {
+      success: response => {
+        resolve(response);
+      },
+      error: (xhr) => {
+        if (xhr.status === 429) {
+          if (timeLeftMs > 0) {
+            console.warn("Request failed with status code 429, message " +
+                new ErrorWrapper(xhr).getUserVisibleError() + ". Retrying after " +
+                sleepMs + " ms. On additional 429 errors, will continue to retry for up " +
+                "to " + timeLeftMs + " ms.");
+            // Retry the request, subtracting the current sleep duration from the remaining time
+            // and doubling the sleep duration
+            const newTimeLeft = timeLeftMs - sleepMs;
+            const newSleepMs = Math.min(newTimeLeft, sleepMs * 2);
+            return new Promise(resolveRetry => setTimeout(resolveRetry, sleepMs)).then(() => {
+              return wrapDeferred(deferred, data, newTimeLeft, newSleepMs);
+            }).then(
+                (successResponse) => resolve(successResponse),
+                (failureResponse) => reject(failureResponse)
+            );
+          }
+        }
         console.error("XHR failed", xhr);
         // We can't throw the XHR itself because it looks like a promise to the
         // redux-promise-middleware.
-        reject(new ErrorWrapper(xhr));
+        return reject(new ErrorWrapper(xhr));
```

Comment:
Had to add this to fix a linter error, even though the return doesn't do anything.

Reply:
Which linter error? This code has been there for some time.

Reply:
True, it's probably because I have a different linter version than travis...